cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NotificationSubscriberService.java

```diff
@@ -186,7 +156,7 @@ private String loadMessageId() {
         return Transactions.execute(transactional, new TxCallable<String>() {
           @Override
           public String call(DatasetContext context) throws Exception {
-            return getJobQueue(context).getMessageId(topic);
+            return Schedulers.getJobQueue(context, datasetFramework).getMessageId(topic);
```

Comment:
It is using the `DatasetContext` that the `Transactions#execute` method is exposing.